src/awkward/contents/emptyarray.py

```diff
@@ -364,11 +368,12 @@ def _to_arrow(
         if options["emptyarray_to"] is None:
             return pyarrow.Array.from_buffers(
                 ak._connect.pyarrow.to_awkwardarrow_type(
-                    pyarrow.null(),
-                    options["extensionarray"],
-                    options["record_is_scalar"],
-                    mask_node,
-                    self,
+                    storage_type=pyarrow.null(),
+                    use_extensionarray=options["extensionarray"],
+                    record_is_scalar=options["record_is_scalar"],
+                    mask=mask_node,
+                    node=self,
+                    is_nonnullable_nulltype=True,
```

Comment:
You don't want this to be `True` for every EmptyArray—you want it to be `True` for those EmptyArrays that are artificially wrapped in UnmaskedArray. If the EmptyArray had already been nested within some other option type, you want `is_nonnullable_nulltype=False`, so that you don't turn what _should_ be `?unknown` into `unknown` when reading back from Arrow.

Reply:
You're absolutely right. I'm changing this to `mask_node is None`. Yet again I wish I had a rock-solid understanding of what I'm doing here, but this *feels* right. If a mask is provided, then you're inside an option type. Unless maybe you're in an UnmaskedArray? Argh! I'm not unit-testing yet.